### ISSUE TITLE
Can't install Version 1.8.5 on Ubuntu 22.04.3 LTS

### ISSUE THREAD

**Issue by HJKLE** [typora/typora-issues#5953]
- [x] Searched existing issues to avoid creating duplicates.

- [x] Confirmed that it can be reproduced in built-in themes without customized css.

- [x] Searched http://support.typora.io/

----

**Describe the bug**

Can't install version 1.8.5 (AMD-Version, downloaded as deb) on Ubuntu 22.04.3 LTS. `apt` reports missing dependencies for `gvfs-bin` and `python`.

Problem 1: `python` is already installed (3.10.6-1~22.04)
Problem 2: `gvfs-bin` deprecated and removed since 2022-02-15 ([Link](https://ubuntu.pkgs.org/22.04/ubuntu-updates-main-amd64/gvfs-common_1.48.2-0ubuntu1_all.deb.html))

**To Reproduce**

1. Download the DEB package from https://typora.io/releases/all
2. Try to install: `sudo dpkg -i typora_1.8.5_amd64.deb`
3. See error

**Expected behavior**

An installation of the package without errors.

**Desktop (please complete the following information):**

 - OS: Ubuntu 22.04.3 LTS

**Typora Version**

e.g: 1.8.5

**Additional Informations**

Version 1.7.6. works properly so far.

**Comment by aizelauna** [comment]
Same issue here:
```
$ sudo aptitude install typora=1.8.5-1
Les NOUVEAUX paquets suivants vont être installés :     
  gconf-service{a} gconf-service-backend{a} gconf2{a} gconf2-common{a} libgconf-2-4{a} 
Les paquets suivants seront mis à jour : 
  typora{b} 
1 paquets mis à jour, 5 nouvellement installés, 0 à enlever et 0 non mis à jour.
Il est nécessaire de télécharger 91,3 Mo d'archives. Après dépaquetage, 31,2 Mo seront utilisés.
Les paquets suivants ont des dépendances non satisfaites :
 typora : Dépend: gvfs-bin which is a virtual package and is not provided by any available package

          Dépend: python which is a virtual package and is not provided by any available package
```

Language is in french, but I think important stuff should be uncipherable by english readers.

**Comment by MagnusPGBerg** [comment]
This is the same as #5954. I just downloaded a deb package and installed Typora 1.8.5 with dpkg -i on Devuan unstable and Gnome 44. It runs fine - yes, it does, started from a terminal - but the package manager said it can't configure Typora and asks for git, gconf2, gconf-service, gvfs-bin and python. I think it's just some errors in the deb packages files.

In the Typora 1.8.5 deb packages file INFO it's written:
_Depends: git, gconf2, gconf-service, gvfs-bin, libc6, libcap2, libgtk2.0-0, libudev0 | libudev1, libgcrypt11 | libgcrypt20, libnotify4, libnss
Recommends: lsb-release
Suggests: pandoc, libgnome-keyring0, lsb-release_

In the Typora 1.7.6 deb packages file INFO it's **only** written:
_Recommends: lsb-release
Suggests: pandoc_
No Depends!

**Remove all Depends, and from Suggests remove libgnome-keyring0, lsb-release, from the INFO file and everything will be all right. I think...**

**Comment by A-Joshi** [comment]
Same issue here. I am able to work around it (I think) by creating two dummy packages gvfs-bin and python. I created the dummy debian packages using the script below (remember to either symlink with the name of the dummy package or create copies of the script with the name of the dummy package).

create the dummy-package.sh (as below).
ln -s dummy-package.sh gvfs-bin
ln -s dummy-package.sh python
./gvfs-bin
./python
sudo apt install `pwd`/gvfs_99_all.deb `pwd`/python_99_all.deb

File dummy-package.sh
```#! /bin/bash

# Idea based on the post at https://fekir.info/post/create-dummy-deb-packages/

SCRIPT_NAME=`basename ${0}`
SCRIPT_DIR=`dirname ${0}`
SCRIPT_RDIR=`(cd ${SCRIPT_DIR} ; pwd -P)`

function Echo() {
    local msg=${*}
    local date=`date`
    echo "# ${SCRIPT_NAME} ${date}: ${msg}"
}

function Exit() {
    local rc=${1}
    shift
    local msg=${*}
    if [ ${rc} -ne 0 ] ; then
       Echo "${msg}. RC=${rc}"
       exit ${rc}
    fi
}

package=${SCRIPT_NAME}

cat << EOF > anti-${package}.equivs
Package: ${package}
Version: 99:99
Maintainer: Your Name <mail@domain.com>
Architecture: all
Description: dummy ${package} package
 A dummy package with a version number so high that the real gnome packages
 will never reach it.
EOF
equivs-build anti-${package}.equivs;
```

**Comment by abnerlee** [comment]
I will revert its depends and suggestions to 1.7.6, current depends is https://github.com/electron-userland/electron-installer-debian/blob/53fb5c500f334fd0e7ea142532f0dcff6728ed7c/src/dependencies.js

**Comment by johnblommers** [comment]
On PopOS 22.04 likewise:

```bash
sudo gdebi typora_1.8.5_amd64.deb 
[sudo] password for XXXXXXXXX: 
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Reading state information... Done
This package is uninstallable
Dependency is not satisfiable: gvfs-bin
```

**Comment by miostreams** [comment]
Problem persists on 1.8.6, with Ubuntu 23.10 (#5954)

**Comment by MagnusPGBerg** [comment]
miostreams
The Depends is still the same in 1.8.6 with a few minor changes:

In the Typora 1.8.6 deb packages file INFO it's written:
```
Depends: git, gconf2, gconf-service, gvfs-bin, libc6, libcap2, libgtk2.0-0, li>
Recommends: lsb-release
Suggests: pandoc
```

In the Typora 1.8.5 deb packages file INFO it's written:
```
Depends: git, gconf2, gconf-service, gvfs-bin, libc6, libcap2, libgtk2.0-0, libudev0 | libudev1, libgcrypt11 | libgcrypt20, libnotify4, libnss
Recommends: lsb-release
Suggests: pandoc, libgnome-keyring0, lsb-release
```

In the Typora 1.7.6 deb packages file INFO it's only written:
```
Recommends: lsb-release
Suggests: pandoc
```
**No Depends at all in 1.7.6.**

**Comment by miostreams** [comment]
I thought from the thread, that the depends list would be reverted to 1.7.6.

**Comment by johnblommers** [comment]
> I will revert its depends and suggestions to 1.7.6, current depends is https://github.com/electron-userland/electron-installer-debian/blob/53fb5c500f334fd0e7ea142532f0dcff6728ed7c/src/dependencies.js

Please do so at your earliest convenience, Abner. We are anxous to try out the latest features, especially admonitions/callouts. Thank you :-)

**Comment by mlgoodson** [comment]
Glad this has been reported. Ubuntu 23.10 also has the same issue.  Looking forward to a solution.

**Comment by rebelga** [comment]
Same problem on KDE neon 5.27 (Ubuntu 22.04).

**Comment by MagnusPGBerg** [comment]
Puh... Now I have learned [how to do it](https://www.baeldung.com/linux/package-deb-change-repack).  It was in the file /DEBIAN/control, in the package, the changes should be done.
I made a package the way I learned in the link, and now it works as it should. You can find my package, and try it, [here](https://motbild.se/files/typora_1.8.6-1+nodependency_amd64.deb). No dependency at all. Just:
```
 Recommends: lsb-release
 Suggests: pandoc
```

**Comment by miostreams** [comment]
Thanks, @MagnusPGBerg !
I built my own .deb, using the instructions you provided. Good info!

**Comment by joshcangit** [comment]
Also got this issue too.
So `gvfs-bin` is no longer installable.
It has been replaced with `gio` commands from `libglib2.0-bin`.

Similar to this.
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=877747

```
$ grep replacement= /usr/bin/gvfs-*
/usr/bin/gvfs-cat:replacement="gio cat"
/usr/bin/gvfs-copy:replacement="gio copy"
/usr/bin/gvfs-info:replacement="gio info"
/usr/bin/gvfs-ls:replacement="gio list"
/usr/bin/gvfs-mime:replacement="gio mime"
/usr/bin/gvfs-mkdir:replacement="gio mkdir"
/usr/bin/gvfs-monitor-dir:replacement="gio monitor"
/usr/bin/gvfs-monitor-file:replacement="gio monitor"
/usr/bin/gvfs-mount:replacement="gio mount"
/usr/bin/gvfs-move:replacement="gio move"
/usr/bin/gvfs-open:replacement="gio open"
/usr/bin/gvfs-rename:replacement="gio rename"
/usr/bin/gvfs-rm:replacement="gio remove"
/usr/bin/gvfs-save:replacement="gio save"
/usr/bin/gvfs-set-attribute:replacement="gio set"
/usr/bin/gvfs-trash:replacement="gio trash"
/usr/bin/gvfs-tree:replacement="gio tree"
```

If Typora relies on any of these commands, they should either be updated or have aliases made.

**Comment by johnblommers** [comment]
**Update**: This method leaves Typora unable to export with Pandoc. The editor itself works fine. Best to wait for Abner Lee to update the repository.

---

Here is a simple workaround to get Typora 1.8.6 installed:

```bash
apt-get download typora
sudo dpkg -i typora_1.8.6-1_amd64.deb 
```

Ignore any warnings about  `gvfs-bin`  and `python` not being installed and enjoy the new Typora. Here is how it goes down:

```bash
apt-get download typora
Get:1 https://typora.io/linux ./ typora 1.8.6-1 [90.4 MB]
Fetched 90.4 MB in 10s (9,169 kB/s)                                                                                    
user@host:~/Desktop$ ls *deb
typora_1.8.6-1_amd64.deb
user@host:~/Desktop$ sudo dpkg -i typora_1.8.6-1_amd64.deb 
(Reading database ... 2060897 files and directories currently installed.)
Preparing to unpack typora_1.8.6-1_amd64.deb ...
Unpacking typora (1.8.6-1) over (1.7.6-1) ...
dpkg: dependency problems prevent configuration of typora:
 typora depends on gvfs-bin; however:
  Package gvfs-bin is not installed.
 typora depends on python; however:
  Package python is not installed.

dpkg: error processing package typora (--install):
 dependency problems - leaving unconfigured
Processing triggers for bamfdaemon (0.5.6+22.04.20220217-0ubuntu1) ...
Rebuilding /usr/share/applications/bamf-2.index...
Processing triggers for mailcap (3.70+nmu1ubuntu1) ...
Processing triggers for gnome-menus (3.36.0-1ubuntu3) ...
Processing triggers for mate-menus (1.26.0-2ubuntu2) ...
Processing triggers for desktop-file-utils (0.26-1ubuntu3) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Errors were encountered while processing:
 typora
```

**Comment by miostreams** [comment]
The latest build resolves this issue: 1.8.7

**Comment by abnerlee** [comment]
fixed in 1.8.8

**Comment by johnblommers** [comment]
Thanks for the quick fix!
